### PR TITLE
Added functionality to edit Question Options

### PR DESCRIPTION
**Pull request by wuergrob**
### General

- Added a "Edit" Button to edit Question Options
- Added a "Save" and a "Discard" Button to manage edits
- Pressing "Enter" will save changes in the form
- Pressing "Escape" will discard the changes

## Related Issues
Concernes: "Improved & accessible SC/MC editor"
https://github.com/uzh-bf/klicker-uzh/issues/3

### CODE REVIEW

File: src/components/questionTypes/SC/SCCreationOption.js [hunk 1]
```diff
@@ -11,13 +12,25 @@ const propTypes = {
   handleCorrectToggle: PropTypes.func.isRequired,
   handleDelete: PropTypes.func.isRequired,
   name: PropTypes.string.isRequired,
+  toggleEditMode: PropTypes.func.isRequired,
```
Comment (rschlaefli) [inline src/components/questionTypes/SC/SCCreationOption.js]: Please name all callbacks `handle...` as is done for the others (just to stick to a convention)

File: src/components/questionTypes/SC/SCCreationOptions.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ const propTypes = {
   handleUpdateOrder: PropTypes.func.isRequired,
   invalid: PropTypes.bool.isRequired,
   value: PropTypes.arrayOf(PropTypes.object).isRequired,
+  saveNewName: PropTypes.func.isRequired,
```
Comment (rschlaefli) [inline src/components/questionTypes/SC/SCCreationOptions.js]: handle..

File: src/components/questionTypes/SC/SCEditQuestionOption.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      const newName = name
+      saveNewName({ newName })
+    },
+
+    discardNewName: ({ setEditMode, origName, setName }) => () => {
+      setEditMode(false)
+      setName(origName)
+    },
+
+    handleKeyPress: ({ name, setEditMode, setName, setOrigName, origName, saveNewName }) => keypress => {
```
Comment (rschlaefli) [inline src/components/questionTypes/SC/SCEditQuestionOption.js]: I would move this to a separate withHandlers after the current one, so you can make use of the saveNewName and discardNewName you have just defined.

File: src/components/questionTypes/SC/SCEditQuestionOption.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    </div>
+    <style jsx>{styles}</style>
+  </div>
+)
+
+SCEditQuestionOption.propTypes = propTypes
+
+export default compose(
+  withState('editMode', 'setEditMode', false),
+  withState('name', 'setName', props => {
```
Comment (rschlaefli) [inline src/components/questionTypes/SC/SCEditQuestionOption.js]:
```suggestion
  withState('name', 'setName', props => props.name)
```

File: src/components/questionTypes/SC/SCEditQuestionOption.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    return props.name
+  }),
+  withState('origName', 'setOrigName', props => {
```
Comment (rschlaefli) [inline src/components/questionTypes/SC/SCEditQuestionOption.js]:
```suggestion
  withState('origName', 'setOrigName', props => props.name)
```

File: src/components/questionTypes/SC/SCEditQuestionOption.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        </button>
+      ) : null}
+    </div>
+    <style jsx>{styles}</style>
+  </div>
+)
+
+SCEditQuestionOption.propTypes = propTypes
+
+export default compose(
```
Comment (rschlaefli) [inline src/components/questionTypes/SC/SCEditQuestionOption.js]: You could generally look at `withStateHandlers` for this logic, as it allows to define state and handlers in the same `withStateHandlers` HOC. And finally have keypress in a subsequent withHandlers. That should reduce the number of needed HOCs to only two.

File: src/components/questionTypes/SC/SCCreationOptions.js [hunk 7]
```diff
@@ -131,5 +137,10 @@ export default compose(
 
     handleUpdateOrder: ({ onChange, value }) => ({ oldIndex, newIndex }) =>
       onChange(arrayMove(value, oldIndex, newIndex)),
+
+    saveNewName: ({ value }) => index => ({ newName }) => {
+      const option = value[index]
+      option.name = newName
```
Comment (rschlaefli) [inline src/components/questionTypes/SC/SCCreationOptions.js]: This actually works? Interesting

**Comment by wuergrob** [comment]
Updated Branch with requested changes and missing features.

**Pull request merged**